kDriveCore/VideoPlayer/VideoPlayer.swift

```diff
@@ -0,0 +1,122 @@
+
+    public lazy var playerViewController: AVPlayerViewController = {
+        let playerViewController = AVPlayerViewController()
+        playerViewController.player = self.player
+        return playerViewController
+    }()
+
+    public init(frozenFile: File, driveFileManager: DriveFileManager) {
+        setupPlayer(with: frozenFile, driveFileManager: driveFileManager)
+    }
+
+    public func setNowPlayingMetadata() {
+        var nowPlayingInfo = [String: Any]()
+        nowPlayingInfo[MPNowPlayingInfoPropertyMediaType] = MPNowPlayingInfoMediaType.video.rawValue
+        nowPlayingInfo[MPNowPlayingInfoPropertyIsLiveStream] = false
+
+        if let currentTrackMetadata {
+            nowPlayingInfo[MPMediaItemPropertyTitle] = currentTrackMetadata.title
+            nowPlayingInfo[MPMediaItemPropertyArtist] = currentTrackMetadata.artist
+            if let artwork = currentTrackMetadata.artwork {
+                let artworkItem = MPMediaItemArtwork(boundsSize: artwork.size) { _ in artwork }
+                nowPlayingInfo[MPMediaItemPropertyArtwork] = artworkItem
+            }
+        } else {
+            nowPlayingInfo[MPMediaItemPropertyTitle] = playableFileName ?? ""
+        }
+
+        if let duration = player?.currentItem?.duration {
+            nowPlayingInfo[MPMediaItemPropertyPlaybackDuration] = CMTimeGetSeconds(duration)
+        }
+
+        MPNowPlayingInfoCenter.default().nowPlayingInfo = nowPlayingInfo
+    }
+
+    public func stopPlayback() {
+        player?.pause()
+    }
+
+    private func setupPlayer(with file: File, driveFileManager: DriveFileManager) {
+        playableFileName = file.name
+
+        if !file.isLocalVersionOlderThanRemote {
+            player = AVPlayer(url: file.localUrl)
+            Task { @MainActor in
+                currentTrackMetadata = await MediaMetadata.extractTrackMetadata(from: file, title: playableFileName)
+            }
+        } else if let token = driveFileManager.apiFetcher.currentToken {
+            driveFileManager.apiFetcher.performAuthenticatedRequest(token: token) { token, _ in
+                if let token = token {
+                    let url = Endpoint.download(file: file).url
+                    let headers = ["Authorization": "Bearer \(token.accessToken)"]
+                    let asset = AVURLAsset(url: url, options: ["AVURLAssetHTTPHeaderFieldsKey": headers])
+                    Task { @MainActor in
+                        self.player = AVPlayer(playerItem: AVPlayerItem(asset: asset))
+                        self.addPeriodicTimeObserver()
+                    }
+                }
+            }
+        }
+    }
+
+    private func addPeriodicTimeObserver() {
+        guard let player = player else { return }
+        let interval = CMTime(seconds: 1, preferredTimescale: CMTimeScale(NSEC_PER_SEC))
+        player.addPeriodicTimeObserver(forInterval: interval, queue: .main) { [weak self] _ in
+            self?.updateNowPlayingInfo()
+        }
+    }
```

Comment:
There is probably a better way to connect AVPlayer's time to the MPNowPlayingInfoCenter

Reply:
I'll check the PR again after this is done.

Reply:
We checked with @Matthieu-dgl The alternative solution seems to be using KVO.

Reply:
I took some time to spend on this issue.
Yeah unlike for an audio track where we need to track progress for our _own_ view, we can do without KVO and without a `PeriodicTimeObserver`. Also for the audio player, a nice to have could be to use a `CADisplayLink` instead.